src/tyr/height_serializer.cc

```diff
@@ -1,60 +1,80 @@
-#include "baldr/json.h"
 #include "skadi/sample.h"
 #include "tyr/serializers.h"
-
-#include <sstream>
+#include "valhalla/baldr/rapidjson_utils.h"
 
 using namespace valhalla;
 using namespace valhalla::midgard;
 using namespace valhalla::baldr;
 
 namespace {
 
-json::ArrayPtr serialize_range_height(const std::vector<double>& ranges,
-                                      const std::vector<double>& heights,
-                                      const uint32_t precision,
-                                      const double no_data_value) {
-  auto array = json::array({});
+void serialize_range_height(rapidjson::writer_wrapper_t& writer,
+                            const std::vector<double>& ranges,
+                            const std::vector<double>& heights,
+                            const uint32_t precision,
+                            const double no_data_value) {
+  writer.start_array("range_height");
+  // precisoin is sent with value 0 which breaks the condition: maxDecimalPlaces >= 1 and causes an
+  // error
+  writer.set_precision(std::max(precision, 1u));
   // for each posting
   auto range = ranges.cbegin();
 
   for (const auto height : heights) {
-    auto element = json::array({json::fixed_t{*range, 0}});
+    writer.start_array();
+
+    if (precision == 0) {
+      // cast to integer after rounding instead of changing tests in valhalla/test/skadi_service.cc
```

Comment:
these kind of comments should be done in the PR, not the code. it's kinda obvious what's happening here. side note: we don't just want the tests to pass, it's much more important to not change service behavior even if tests pass. amounts to the same thing in this case, just noting.

since we seem to do 3 times the exact same, maybe it's better to offload into a small lamdba whether it writes an int or a float.